repo-agent/llm/gemini.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package llm
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+// Gemini is an Provider that uses the gemini-cli.
+//
+// Make sure that the Gemini struct implements the Provider interface.
+var _ Provider = &Gemini{}
+
+type Gemini struct {
+	Executor CommandExecutor
```

Comment:
The `temperature` is hardcoded to `1.0`. This is another critical parameter for controlling model creativity that should be configurable.